Server/src/Config/Config.java

```diff
@@ -65,4 +67,66 @@ public String getProperty(String property) {
     return config.getProperty(property);
   }
 
+  private Properties loadDefaultConfigFile() {
+    Properties defaultProperties = null;
+
+    try {
+      InputStream defaultConfig = Config.class.getResourceAsStream(configFileName);
+      if (defaultConfig == null) {
+        throw new FileNotFoundException("Couldn't find default configuration at " + configFileName);
+      }
+      defaultProperties = new Properties();
+      defaultProperties.load(defaultConfig);
+
+      defaultConfig.close();
+    } catch (FileNotFoundException ex) {
+      ex.printStackTrace();
+    } catch (IOException ex) {
+      ex.printStackTrace();
+    }
+
+    return defaultProperties;
+  }
+
+  public boolean shouldUpdateUserConfig() {
+    boolean shouldUpdateUserConfig = false;
+    try {
+      if (config == null) {
+        throw new Exception("User's config file is null");
+      }
+
+      Properties defaultConfig = loadDefaultConfigFile();
+      Set<String> defaultVariables = defaultConfig.stringPropertyNames();
+      Set<String> userVariables = config.stringPropertyNames();
+      if (!defaultVariables.equals(userVariables)) {
+        shouldUpdateUserConfig = true;
+      }
+    } catch (Exception ex) {
+      ex.printStackTrace();
+    }
+
+    return shouldUpdateUserConfig;
+  }
+
+  public void updateUserConfig() {
+
+    Properties defaultConfig = loadDefaultConfigFile();
+    Properties newValuesToAppend = new Properties();
+    defaultConfig.forEach((key, value) -> {
+      if (!config.containsKey(key.toString())) {
+        newValuesToAppend.put(key.toString(), value.toString());
+        System.out.println("added " + key.toString());
+      }
+    });
+
+    try {
+      AppDirs appDirs = AppDirsFactory.getInstance();
+      String configDirPath = appDirs.getUserConfigDir("cardboard", "1.0", "projectcardboard", true);
+      Path configFullPath = Path.of(configDirPath + configFileName);
+      OutputStream out = new FileOutputStream(configFullPath.toString(), true);
+      newValuesToAppend.store(out, "\n");
```

Comment:
Instead of the whole `newValuesToAppend` thing, could you just `put` them into `config` (which is already a loaded Properties file) then call `config.store(out, "\n")`?

Reply:
config.store(out, "\n") writes all of the key value pairs stored inside the properties table to the file. So in the event we only add one new variable, we will write the new variable and all pre existing ones into the file. I want to only add the variables the user is missing so i went this appraoch

Reply:
Change `new FileOutputStream(configFullPath.toString(), false)` and that's not a problem. Sure we'll be technically rewriting the whole file, but with the contents that were there already plus the new stuff.